nasdaqdatalink/model/authorized_session.py

```diff
@@ -0,0 +1,57 @@
+import nasdaqdatalink
+from nasdaqdatalink.api_config import ApiConfig
+from urllib3.util.retry import Retry
+from requests.adapters import HTTPAdapter
+import requests
+import urllib
+
+
+def get_retries(api_config=nasdaqdatalink.ApiConfig):
+    retries = None
+    if not api_config.use_retries:
+        return Retry(total=0)
+
+    Retry.BACKOFF_MAX = api_config.max_wait_between_retries
+    retries = Retry(total=api_config.number_of_retries,
+                    connect=api_config.number_of_retries,
+                    read=api_config.number_of_retries,
+                    status_forcelist=api_config.retry_status_codes,
+                    backoff_factor=api_config.retry_backoff_factor,
+                    raise_on_status=False)
+    return retries
+
+
+class AuthorizedSession:
+    def __init__(self, api_config=ApiConfig) -> None:
+        super(AuthorizedSession, self).__init__()
+        if not isinstance(api_config, ApiConfig):
+            api_config = ApiConfig
+        self._api_config = api_config
+        self._auth_session = requests.Session()
+        retries = get_retries(self._api_config)
+        adapter = HTTPAdapter(max_retries=retries)
+        self._auth_session.mount(api_config.api_protocol, adapter)
+
+        proxies = urllib.request.getproxies()
+        if proxies is not None:
+            self._auth_session.proxies.update(proxies)
+
+    def get(self, dataset, **kwargs):
```

Comment:
Outside of this tickets scope, but I would like to see us passing connection details (session, api_config) as explicit arguments, and leave kwargs to params we should send to the API call. This would allow us to avoid all the kwarg popping down the line